library/kotlin/src/io/envoyproxy/envoymobile/RequestMethod.kt

```diff
@@ -1,15 +1,33 @@
 package io.envoyproxy.envoymobile
 
+import java.lang.IllegalArgumentException
+
 /**
  * Represents an HTTP request method.
  */
-enum class RequestMethod {
-  DELETE,
-  GET,
-  HEAD,
-  OPTIONS,
-  PATCH,
-  POST,
-  PUT,
-  TRACE
+enum class RequestMethod(internal val stringValue: String) {
+  DELETE("DELETE"),
+  GET("GET"),
+  HEAD("HEAD"),
+  OPTIONS("OPTIONS"),
+  PATCH("PATCH"),
+  POST("POST"),
+  PUT("PUT"),
+  TRACE("TRACE");
+
+  companion object {
+    fun enumValue(stringRepresentation: String): RequestMethod {
```

Comment:
I think this can be internal

Reply:
I see where you are coming from. Everything outside of the module should deal with only enum types, not strings. I like that.